network/horizon/api-reference/errors/error-handling.mdx

```diff
@@ -0,0 +1,184 @@
+
+## Error Handling for Queries
+
+Many `GET` requests have specific parameter requirements, and while the SDKs can help enforce them, you can still pass invalid arguments (for example, an asset string that isn’t SEP-11 compatible) that error out every time. In this scenario, there’s nothing you can do aside from following the API specification. The `extras` field of the error response will often clue you in on where to look and what to look for.
+
+<CodeExample>
+
+```bash
+curl -s https://horizon-testnet.stellar.org/claimable_balances/0000 | jq '.extras'
+{
+    "invalid_field": "id",
+    "reason": "Invalid claimable balance ID"
+}
+```
+
+</CodeExample>
+
+Note that the SDKs make it a point to distinguish an invalid request (as above) versus a missing resource (a `404 Not Found`) (for example, the generic `NetworkError` versus a `NotFoundError` in the JavaScript SDK), where the latter might not be considered an error depending on your situation.
+
+## Error Handling for Transaction Submissions
+
+Horizon currently supports two types of transaction submission endpoints:
+
+1. `/transactions_async`: Horizon submits a transaction to Stellar-Core in an asynchronous manner and passes the relevant Stellar-Core response back immediately to the client. It is then the client's responsibility to poll for the status of that transaction.
+2. `/transactions`: Horizon submits a transaction to Stellar-Core and then waits for it to be ingested in the database. It will either timeout while waiting or return a transaction response on success.
+
+There are some resolution strategies that are common between the 2 endpoints while some strategies are more endpoint specific.
+
+### Request Adjustments
+
+Certain transaction submission failures also need adjustments to succeed.
+
+- If the XDR is malformed, or the transaction is otherwise invalid, you’ll encounter a `400 Bad Request` (for example, see excluding a source account). Both transactions and their operations can be easily malformed: look at the `extras.result_codes` field for details and cross-reference them with the appropriate result codes documentation to determine specifics.
+- Transaction fees are also a safe adjustment. If you get a `tx_insufficient_fee` error, refer to the Insufficient Fees and Surge Pricing section later in this document.
+
+### Polling and Retrying Transactions
+
+#### Async Transaction Submission
+
+Submissions using the `/transactions_async` endpoint return an immediate response back from Stellar-Core. There are different actions that clients can take based on the specific `tx_status` returned:
+
+1. `PENDING`: The submission is successful but the transaction is still waiting to be included in a ledger. You should use the [GET /transactions/:transaction_hash](/network/horizon/api-reference/resources/retrieve-a-transaction) endpoint to poll for the submitted transaction and check if it makes into a ledger.
```

Comment:
You should clarify that it can still fail at this point despite submission being successful :+1:

Reply:
Ah nice catch. Yeah I will add that